internal/http/services/owncloud/ocdav/move.go

```diff
@@ -217,6 +217,20 @@ func (s *svc) handleMove(ctx context.Context, w http.ResponseWriter, r *http.Req
 			w.WriteHeader(http.StatusPreconditionFailed) // 412, see https://tools.ietf.org/html/rfc4918#section-9.9.4
 			return
 		}
+		// do not allow overwriting spaces
+		// check if it is a root?
+		if dstStatRes.GetInfo().GetPath() == "." {
+			log.Warn().Bool("overwrite", overwrite).Msg("overwriting spaces via dav is not allowed ")
+			w.WriteHeader(http.StatusPreconditionFailed) // 412, see https://tools.ietf.org/html/rfc4918#section-9.9.4
+			return
+		}
+		// validate if the destination can be overviewed
+		if dstStatRes.GetInfo().GetPath() != "" && dstStatRes.GetInfo().GetMimeType() != "" &&
+			(srcStatRes.GetInfo().GetPath() != dstStatRes.GetInfo().GetPath() || srcStatRes.GetInfo().GetType() != dstStatRes.GetInfo().GetType()) {
```

Comment:
I don't quite get this logic. What are we trying to check here?

Also I think `srcStatRes.GetInfo().GetType() != dstStatRes.GetInfo().GetType()` is wrong. As discussed in standup we want to allow moving a file to a folder

Reply:
It is the wrong destination I think. Because the destination for a file has to be a file regarding the specs
http://www.webdav.org/specs/rfc4918.html#n-example---move-of-a-non-collection